packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.ts

```diff
@@ -434,6 +488,9 @@ export default {
             node,
             reference.identifier,
           );
+          if (!referenceNode) {
+            continue;
+          }
```

Comment:
Why is this needed?

Reply:
`fastFindReferenceWithParent` returns null if it's unable to find a suitable node, but all the logic below this assumes that it's returned a Node.  In practice, it shouldn't really ever not return a node, so this is more of an edge case, but since the function is narrowing out null, this early return keeps the rest of the assumptions in tact.